### ISSUE TITLE
`.getMatrix` on `Canvas`

### ISSUE THREAD

**Issue by harold** [JetBrains/skija#97]
Hi! Thanks so much for skija, it's sped up my rendering at least 4x over what I was using before, and that's just great!

`.setMatrix` on `Canvas` (as well as `concat`) is super-helpful in the implementation of my renderer.
https://github.com/JetBrains/skija/blob/master/shared/src/main/java/org/jetbrains/skija/Canvas.java#L925-L930

Would it be possible to get something like `.getMatrix` to retrieve the current `Matrix33` for the Canvas?

I found `.getLocalToDevice` (https://github.com/JetBrains/skija/blob/master/shared/src/main/java/org/jetbrains/skija/Canvas.java#L950-L958), but that returns a `Matrix44`.

My renderer has the notion of hierarchical contexts and being able to push/pop a stack of matrix transforms would be lovely. I can implement this myself with `setMatrix` and `resetMatrix`, but it might be nice to leverage `concat` and just be able to query the `Matrix33` that skia has to avoid confusion.

Regardless, thank you for your time and consideration - the library is wonderful. Keep it up.

Enjoy the attached video, rendered with skija this morning!

https://user-images.githubusercontent.com/7443/110151060-3a840480-7d9d-11eb-9996-a8dc1ed1ab6a.mp4

@cnuernber - again, perhaps relevant to your interests.

**Comment by tonsky** [comment]
This is what Skia has:

```
    /** Returns the current transform from local coordinates to the 'device', which for most
     *  purposes means pixels.
     *
     *  @return transformation from local coordinates to device / pixels.
     */
    SkM44 getLocalToDevice() const;

    /**
     *  Throws away the 3rd row and column in the matrix, so be warned.
     */
    SkMatrix getLocalToDeviceAs3x3() const {
        return this->getLocalToDevice().asM33();
    }

#ifdef SK_SUPPORT_LEGACY_GETTOTALMATRIX
    /** DEPRECATED
     *  Legacy version of getLocalToDevice(), which strips away any Z information, and
     *  just returns a 3x3 version.
     *
     *  @return 3x3 version of getLocalToDevice()
     *
     *  example: https://fiddle.skia.org/c/@Canvas_getTotalMatrix
     *  example: https://fiddle.skia.org/c/@Clip
     */
    SkMatrix getTotalMatrix() const;
#endif
```

I’ll add `getLocalToDeviceAs3x3`, but be warned (as per Skia docs) :)

P.S. Video is DOPE

**Comment by harold** [comment]
> `*  Throws away the 3rd row and column in the matrix, so be warned.`
>

Ha! OK. Will have to experiment to know if that's useful or not.

> P.S. Video is DOPE
>

🙇 - thank you! You may enjoy my Instagram, in that case: https://www.instagram.com/_hrrld/

Stay rad.

**Comment by Solido** [comment]
Video is DOPE !!!

**Comment by harold** [comment]
Here's another. Skija performing well!

https://user-images.githubusercontent.com/7443/110492297-49292f00-80af-11eb-8215-a897648d007f.mp4

**Comment by tonsky** [comment]
Closed in 7b76c0a0942435c3d3d7b5d74ceb1d49f609ceef

**Comment by harold** [comment]
👏
